blog/2023-04-23-json.md

```diff
@@ -0,0 +1,172 @@
+
+JSON has become the de facto standard for data interchange on the web. As a minimal, text-based format that is a subset of JavaScript, JSON is easy to read and focuses on solving a single problem: data interchange. Its simplicity and widespread adoption make JSON a clear choice for incorporating into Wing, helping us achieve the vision of a truly cloud-oriented programming language.
+
+## Introducing Wing's Json
+
+Wing has decided to treat JSON as a first-class citizen with built-in primitives. Let's take a look at a simple example:
+
+```js
+let j = {
+  wow: "so easy"
+};
+
+// Optionally we can also add the Json keyword prefix
+let j = Json {
+  wow: "so easy"
+};
+```
+
+That's all it takes to create a JSON object in Wing. No need to import anything!
+
+Now, let's create an inflight function that requires a Json object as input:
+
+```js
+bring cloud;
+
+let b = new cloud.Bucket();
+
+let someService = new cloud.Function(inflight (jsonString: str) => {
+  let message = Json.parse(jsonString);
+  let id = message.get("id");
+  let content = str.fromJson(message.get("content"));
+  b.put("file-${id}.txt", content);
+});
+
+// Test it out
+test "storing a message" {
+  let payload = Json {
+    id: "abc123",
+    content: "Hello, Json!"
+  };
+  someService.invoke(Json.stringify(payload));
+
+  // retrieve object from bucket and validate content
+  let storedMessage = b.get("file-${str.fromJson(payload.get("id"))}.txt");
+  assert(storedMessage == str.fromJson(payload.get("content")));
+}
+```
+
+Something noteworthy happens above when we call `str.fromJson()`. All fields within a `Json` object are of type `Json`, and Wing does not allow for implicit type conversions from `Json` to other types. Thus, we must explicitly convert the `Json` object to a string using the `fromJson()` function, which exists for all Wing primitive types.
+
+Let's dive a little deeper and see what else Wing's `Json` type offers.
+
+### Converting To Structs
+
+While `Json` objects are great for interoperability, they are not the best for working with data in Wing. This is because `Json` objects are not strongly typed. However, Wing does provide a way to convert `Json` objects to structs. This conversion is done by using the `fromJson()` function, which exists on all structs created in Wing.
+
+Lets take a look at an example
+
+```js
+bring cloud;
+
+// Struct definition of a payload object
+struct Payload {
+  id: str;
+  content: str;
+}
+
+// inflight function that recieves json string representing a payload
+new cloud.Function(inflight(msg: str) => {
```

Comment:
You forgot to assign the `cloud.Function` to a variable so you could invoke it in the test below.
```suggestion
let someFunction = new cloud.Function(inflight(msg: str) => {
```